### ISSUE TITLE
Make System Qubes Run on Minimal Templates

### ISSUE THREAD

**Issue by msmhome** [QubesOS/qubes-issues#8567]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### The solution you'd like

Even though the current implementation does not directly invoke tangible issues, refining system Qubes, such as sys-net, sys-usb, and sys-firewall, to utilize Fedora minimal templates as opposed to the full templates is proposed. Streamlining these VMs aims to diminish the potential attack surface and vulnerability points, optimizing security. While this may nominally increase the ISO size, the impact on the initial unpacking of templates during the post-Anaconda install boot could be mitigated (it already does take a while to finish), especially if configuring system VMs to minimal becomes an optional setting—akin to enabling sys-usb.

### The value to a user, and who that user might be

This enhancement will cater to all users prioritizing a minimized risk exposure and a decluttered user interface.

**Security Improvement**: Users gain a mental assurance from the knowledge that operating on minimized templates inherently presents fewer avenues for potential compromises.

**Simplified User Experience**: The appearance of applications, such as Firefox or Keepass, in your sys-usb, might be slightly disconcerting or unnecessary for some users. A minimized template would reduce such instances, presenting a cleaner, more focused user interface.

**System Efficiency**: Minimizing templates may also inadvertently promote better system performance and resource utilization, especially pertinent for users operating on hardware with limited resources.

**Comment by jamke** [comment]
Do you mean `fedora-*-minimal` templates? These templates **are really minimal**, they have no `qubes-` packages, no network stuff, no pulseaudio and other stuff, because user can use this minimal templates as basis for some offline qube or completely different fedora-based template. Templates for `sys-net`, `sys-usb` require a lot of non-minimal stuff inside after all.

* `minimal` should stay minimal as it is now.
* templates for `sys-net`, `sys-usb` can (should?) be smaller than full `fedora-*` templates and may be adjusted in different ways.

=> So, you are proposing adding additional template, or several templates for system qubes out of the box. Am I right?

**Comment by msmhome** [comment]
> Do you mean `fedora-*-minimal` templates? These templates **are really minimal**, they have no `qubes-` packages, no network stuff, no pulseaudio and other stuff, because user can use this minimal templates as basis for some offline qube or completely different fedora-based template. Templates for `sys-net`, `sys-usb` require a lot of non-minimal stuff inside after all.
> 
> * `minimal` should stay minimal as it is now.
> * templates for `sys-net`, `sys-usb` can (should?) be smaller than full `fedora-*` templates and may be adjusted in different ways.

I agree slightly, but it's hardly any packages required to make a sys-net for instance. See here https://www.qubes-os.org/doc/templates/minimal/#fedora

You can make one right now using that doc and a fedora-*-minimal template. Explore your sys-net's available applications, the full Fedora templates are too bulky and have too many user-oriented applications. 

> => So, you are proposing adding additional template, or several templates for system qubes out of the box. Am I right?

Not really. The minimal templates available are more than enough, and perfectly functional for such tasks.

**Comment by marmarek** [comment]
> > => So, you are proposing adding additional template, or several templates for system qubes out of the box. Am I right?
> 
> Not really. The minimal templates available are more than enough, and perfectly functional for such tasks.

If you install the system with sys-net based on the minimal template, then networking in sys-net will not work before you install those extra packages. And you cannot install packages with broken network connection... You could potentially start sys-net with full template, then install network-related packages, and then switch sys-net to the minimal template, but that's way more steps than just "have sys-net (and other service qubes) based on minimal template". And generally the installation image should be self-contained, requiring network connection to finish installation (download network-related packages in this case) is not an acceptable solution.

**Comment by msmhome** [comment]
@marmarek That make sense, and I think every user would agree to keeping the installation process offline. Do you have any idea of a possible course of action (shipping service templates based off of min, including the package and subsidizing with scripts to install them on minimal qubes, having scripts that strip the full Fedora templates of packages for use in service qubes, etc.)? I'm curious about your opinion and whether there are any empirical concerns with the full Fedora templates being in place. Perhaps it's not too much of a tradeoff for most users considering the nuances of implementation and ISO size.

**Comment by jamke** [comment]
> You can make one right now using that doc and a fedora-*-minimal template. Explore your sys-net's available applications, the full Fedora templates are too bulky and have too many user-oriented applications

I do use only own templates based on `fedora-*-minimal` as I do not use gnome-related stuff, so I know it.

My point is that existing minimal templates are not enough for `sys-usb`, `sys-net`, `sys-firewall`, as I told before. So, what is you proposal exactly?
